### PR TITLE
e2e: e2e_node: refactor getCurrentKubeletConfig

### PR DESCRIPTION
**Pull request by ffromani**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Refactoring to de-dup the getCurrentKubeletConfig helper which is used by both e2e and e2e_node tests. This patch originated because we want to add more topology manager e2e tests, and the function started to proliferate too much. See https://github.com/kubernetes/kubernetes/pull/88107

**Which issue(s) this PR fixes**:
Related to https://github.com/kubernetes/kubernetes/issues/83481

**Special notes for your reviewer**:
This patch works for me running e2e_node tests locally, but I don't have windows machine available.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: test/e2e/framework/kubeletconfig.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
+/*
```
Comment (oomichi) [inline test/e2e/framework/kubeletconfig.go]: Could you implement this as `test/e2e/framework/kubelet/config.go`?

If implementing this as `test/e2e/framework/kubeletconfig.go`, this file is defined as core part of e2e test framework.
This file imports `e2ekubectl` which is defined as subpackage of e2e test framework, and that means invalid dependency as https://github.com/kubernetes/kubernetes/issues/81245
By implementing as ` test/e2e/framework/kubelet/config.go` we can avoid such thing.

Comment (oomichi) [reply test/e2e/framework/kubeletconfig.go]: /cc @oomichi

Comment (fromanirh) [reply test/e2e/framework/kubeletconfig.go]: Sure thing @oomichi , on it

**Comment by fromanirh** [comment]
not sure who can review/approve from windows side @michmike and @PatrickLang (according to the OWNERS file) could you please help?

**Comment by oomichi** [comment]
Thanks for updating.

/ok-to-test

**Comment by oomichi** [comment]
Need to run `./hack/update-bazel.sh` for passing pull-kubernetes-verify job.

**Comment by PatrickLang** [comment]
/assign
Will test on Windows

**Comment by oomichi** [comment]
Thanks for updating.

/lgtm

**Comment by derekwaynecarr** [comment]
/approve

/hold

@PatrickLang when you have a chance to confirm, please remove the hold.

**Review by SataQiu — COMMENTED** [review]
/approve
Nice work!
defer @PatrickLang to confirm and remove the hold

**Comment by PatrickLang** [comment]
/test pull-kubernetes-e2e-aks-engine-azure-windows

I'm not sure the state of that prow job, so I'm scheduling my own local run too

**Comment by PatrickLang** [comment]
/hold cancel

Seems ok - the moved code works. The test case still fails but I'll file a separate issue for that. It's not on testgrid at the moment because we're skipping [slow][serial].

**Comment by PatrickLang** [comment]
Filed #88481 for broken Windows test

**Pull request merged**